src/client/reducers/veteran/index.js

```diff
@@ -3,171 +3,140 @@ import lodashDeep from 'lodash-deep';
 
 import { ENSURE_FIELDS_INITIALIZED, VETERAN_FIELD_UPDATE, UPDATE_SPOUSE_ADDRESS, CREATE_CHILD_INCOME_FIELDS } from '../../actions';
 import { makeField, dirtyAllFields } from '../fields';
-import { pathToData } from '../../store';
 
 // Add deep object manipulation routines to lodash.
 _.mixin(lodashDeep);
 
-// TODO(awong): This structure should reflect a logical data model for a veteran. Currently it
-// mirrors the UI stricture too much.
-
 // TODO: Remove providers and children if checkbox within section is unchecked
 const blankVeteran = {
-  nameAndGeneralInformation: {
-    fullName: {
-      first: makeField(''),
-      middle: makeField(''),
-      last: makeField(''),
-      suffix: makeField(''),
-    },
-    mothersMaidenName: makeField(''),
-    socialSecurityNumber: makeField(''),
-    gender: makeField(''),
-    cityOfBirth: makeField(''),
-    stateOfBirth: makeField(''),
-    dateOfBirth: {
-      month: makeField(''),
-      day: makeField(''),
-      year: makeField(''),
-    },
-    maritalStatus: makeField('')
-  },
-
-  vaInformation: {
-    isVaServiceConnected: makeField(''),
-    compensableVaServiceConnected: makeField(''),
-    receivesVaPension: makeField('')
+  fullName: {
+    first: makeField(''),
+    middle: makeField(''),
+    last: makeField(''),
+    suffix: makeField(''),
   },
-
-  additionalInformation: {
-    isEssentialAcaCoverage: false,
-    facilityState: makeField(''),
-    vaMedicalFacility: makeField(''),
-    wantsInitialVaContact: false
+  mothersMaidenName: makeField(''),
+  socialSecurityNumber: makeField(''),
+  gender: makeField(''),
+  cityOfBirth: makeField(''),
+  stateOfBirth: makeField(''),
+  dateOfBirth: {
+    month: makeField(''),
+    day: makeField(''),
+    year: makeField(''),
   },
-
-  demographicInformation: {
-    isSpanishHispanicLatino: false,
-    isAmericanIndianOrAlaskanNative: false,
-    isBlackOrAfricanAmerican: false,
-    isNativeHawaiianOrOtherPacificIslander: false,
-    isAsian: false,
-    isWhite: false
+  maritalStatus: makeField(''),
+
+  isVaServiceConnected: makeField(''),
+  compensableVaServiceConnected: makeField(''),
+  receivesVaPension: makeField(''),
+
+  isEssentialAcaCoverage: false,
+  facilityState: makeField(''),
+  vaMedicalFacility: makeField(''),
+  wantsInitialVaContact: false,
+
+  isSpanishHispanicLatino: false,
+  isAmericanIndianOrAlaskanNative: false,
+  isBlackOrAfricanAmerican: false,
+  isNativeHawaiianOrOtherPacificIslander: false,
+  isAsian: false,
+  isWhite: false,
+
+  address: {
```

Comment:
I think the way that @joesoftware is setting up the backend, everything that is about a veteran is prefixed with "veteran". So `address` would be `veteranAddress` and `county` would be `veteranCounty`. This way we can differentiate between these fields for a veteran, a spouse or a child.